alacritty_terminal/src/term/cell.rs

```diff
@@ -186,7 +186,7 @@ impl Cell {
     pub fn set_hyperlink(&mut self, hyperlink: Option<Hyperlink>) {
         let should_drop = hyperlink.is_none()
             && self.extra.as_ref().map_or(true, |extra| {
-                !extra.zerowidth.is_empty() || extra.underline_color.is_some()
+                !extra.zerowidth.is_empty() || extra.underline_color.is_none()
             });
```

Comment:
Is the zerowidth logic correct here? Shouldn't it be `zerowidth.is_empty() && underline_color.is_none()`?

We only want to drop if zerowidth and color are both not set, right?